### PR TITLE
Create signup form frontend

### LINKED ISSUE #98: Create Signup Form
image

### PR DESCRIPTION
**Pull request by MaramMs**
image

Relates #98 
Close #98

### CODE REVIEW

File: client/src/App.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+body{
+  background-color:black !important;
```
Comment (nada-abuzaid) [inline client/src/App.css]: The background color is not black, and why do you use `!important`? 
```suggestion
  background-color: #1C1C28;
```

File: client/src/App.js [hunk 2]
```diff
@@ -1,9 +1,12 @@
 import React from "react";
+import Signup from './pages/signUp';
+import './App.css'
+import 'bootstrap/dist/css/bootstrap.min.css'
```
Comment (nada-abuzaid) [inline client/src/App.js]: You should import it in the Signup page not here, and put your style in the end
```suggestion
import 'bootstrap/dist/css/bootstrap.min.css'
import './style.css'
```

File: client/src/App.js [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
 
 function App() {
   return (
-    <div className="App">
-      <h1>Taskillo</h1>
+    <div className="container">
```
Comment (nada-abuzaid) [inline client/src/App.js]: Please, don't delete the general code.
```suggestion
    <div className="App container">
```

File: client/src/pages/signUp.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import axios from 'axios'
+import Form from '../components/Form/Form'
+import '../scss/signup.css'
```
Comment (nada-abuzaid) [inline client/src/pages/signUp.jsx]: Put your CSS in the the signup folder with signup.jsx file.
```suggestion
import './signup.css'
```

File: client/src/pages/signUp.jsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+         <h1 className='text-white mb-5 '>Taskillo</h1>
+         <span className='text-white mb-2'>Start For Free</span>
+         <h2 className='text-white mb-4 '>Create an account</h2>
+       <Form 
+        initialValues={ {username:'', code:'',password:''}}
+        validationSchema= {validationSignUpSchema}
+        onSubmit={handleSubmit} >
+        <FormField  type='text' name='username' placeholder='Enter your username'/>
+        <FormField type='password' name='password' placeholder='Enter your password' />
+        <FormField  type='number' name='code' placeholder='Enter  verification code'/>
```
Comment (nada-abuzaid) [inline client/src/pages/signUp.jsx]:
```suggestion
        <FormField  type='number' name='code' placeholder='Enter verification code'/>
```

File: client/src/components/Form/FormField.jsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import { TextError } from './TextError';
+const FormField = ({ placeholder, name, ...rest }) => {
+  return (
+    <div className='form-group'>
+      <Field
+        id={name}
+        name={name}
+        {...rest}
+        placeholder={placeholder}
+        className='form-control  mb-3 w-50 text-white '
```
Comment (nada-abuzaid) [inline client/src/components/Form/FormField.jsx]: Pass the className as a props.
```suggestion
        className={`form-control  ${className} `}
```

**Review by nada-abuzaid — CHANGES_REQUESTED** [review]

**Comment by karamzomlot** [comment]
> image
> 
> Relates #98 Close #98

File: client/src/pages/SignUp/signUp.jsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import axios from 'axios'
+import Form from '../../components/Form/Form'
+import 'bootstrap/dist/css/bootstrap.min.css'
```
Comment (Nizar7zak) [inline client/src/pages/SignUp/signUp.jsx]: you should import Bootstrap in App file not here.

File: client/src/utils/signUpValidation.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import * as Yup from 'yup';
+const validationSignUpSchema = Yup.object({
+  username:Yup.string().min(3,'Username should be at least 3 char ').max(20,'Username should be at most 20 char ').required('username'),
```
Comment (Nizar7zak) [inline client/src/utils/signUpValidation.js]: you should be consistent, so please instead of .required(username)
.required('username is required')

**Review by Nizar7zak — CHANGES_REQUESTED** [review]

**Review by AmranElmasri — APPROVED** [review]

**Review by karamzomlot — APPROVED** [review]

**Review by Nizar7zak — APPROVED** [review]

**Pull request merged**